### PR TITLE
Show message when a date change cannot be accessed; COUNTRY=cambodia

### PR DESCRIPTION
**Pull request by HarryMytilinaios**
This adds a warning message when a date change cannot be accessed or is not intersecting with all layers' available dates.

### CODE REVIEW

**Comment by wadhwamatic** [comment]
Thanks @HarryMytilinaios. The message appears as expected, but the up-arrow (caret) still moves to the selected date, even if it's not active. We should have the caret persist on the selected date.

**Comment by HarryMytilinaios** [comment]
@wadhwamatic Added the caret to persist in the last active selected date. Please let me know if this is what we want.

**Comment by wadhwamatic** [comment]
@HarryMytilinaios - this looks good, thanks! At some point, it would be good to have the layer not re-render but for the purpose of this issue / PR, I think we're good now. @ericboucher - what do you think?

**Review by jbperidypathtech — COMMENTED** [review]
Hi @HarryMytilinaios, you will find two comments here. I continue discovering the app, do not hesitate to join me on Slack if you are not ok with some comments or you need more details.

File: frontend/src/components/MapView/DateSelector/TimelineItems/index.test.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ const props = {
     },
   ],
   clickDate: () => {},
+  locale: 'en',
```
[the remark this answers is not in the document]
Comment (HarryMytilinaios) [reply frontend/src/components/MapView/DateSelector/TimelineItems/index.test.tsx:19]: @jbperidypathtech if you review the component `DateSelector` you will see that the locale is passed as props in the `TimelineItems` and it is from a ternary operator which would be either `t('date_locale')` or `'en'`. The `t` function is coming from and extended hook which is called `useSafeTranslation` and extends the `i18next` library's `useTranslation` so the locale is something that we grab from the `translations.json` which lives inside the config folder and every each country have its own. Moreover the specific file is a test file. We don't need to pass every prop and see the component reaction. We just want to see that with some props (the required ones) the component will be rendered correctly. If the locale was something else the component would still render correctly.

**Comment by HarryMytilinaios** [comment]
@wadhwamatic . I added an extra commit, which prevents the rerendering of the layers in the map when the selectedDate is not an intersecting one, basically we don't update the url so we don't cause repaints in the map.

**Review by jbperidypathtech — APPROVED** [review]

**Review by ericboucher — APPROVED** [review]

**Pull request merged**